core/src/main/java/org/apache/calcite/plan/RelTrait.java

```diff
@@ -85,4 +87,14 @@
    * @param planner Planner
    */
   void register(RelOptPlanner planner);
+
+  /**
+   * Applies a mapping to this trait.
+   *
+   * @param mapping   Mapping
+   * @return trait with mapping applied
+   */
+  default <T extends RelTrait> T apply(Mappings.TargetMapping mapping) {
+    return (T) this;
+  }
```

Comment:
In which cases does the returned trait have a different type from this?

Reply:
Ah, the cast seems not necessary. Thanks.